src/uct/ib/base/ib_iface.c

```diff
@@ -448,6 +459,23 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
               self->config.rx_headroom_offset, self->config.rx_payload_offset,
               self->config.rx_hdr_offset, self->config.seg_size);
 
+    /* init device info hash */
+    kh_init_inplace(uct_ib_device_info_hash, &self->info_hash);
```

Comment:
why do we need this hash? the lookup happens only when doing iface_query, this is not fast-path (see optimization gide). also, how does it work that every time an iface is created, the global variable of hash table is re-initialized?